task2 WeatherApp

mustafakamber opened the pull request.

Commit:
feat: Add task2

scotoglu commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/model/City.kt:
```diff
@@ -0,0 +1,15 @@
+
+@Parcelize
+data class City(
+    val cityId: Int,
```
city prefixleri kaldırabilir.

scotoglu commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/util/CityDataSource.kt:
```diff
@@ -0,0 +1,32 @@
+    fun getCities(context : Context): MutableList<City> {
+        return mutableListOf(
+            City(
+                1, "İstanbul",
```
değerleri verirken hangi alana karşılık kullanıldığını da yazarsak daha okunulabilir olur.` id = 1, name = "İstanbul"`

scotoglu commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/util/Extension.kt:
```diff
@@ -0,0 +1,46 @@
+
+object Extension {
+
+    fun Fragment.getFragment(appCompatActivity: AppCompatActivity, bundle: Bundle? = null) {
```
FragmentExt olarak ayrı bir extension dosyası oluşturabiliriz.

iamkurtgoz replied:
ismi route, navigate prefix olabilir

scotoglu commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/util/Extension.kt:
```diff
@@ -0,0 +1,46 @@
+        transaction.commit()
+    }
+
+    fun Activity.getActivity(destinationActivity: Activity,city : City? = null){
```
ActivityExt olarak ayrı bir extension dosyası oluşturabiliriz.

iamkurtgoz replied:
startActivity olabilir, 

fun Activity.startActivity(destinationActivity: Activity, city: City? = null){

akndmr replied:
İçeriğine bakınca yaptığı işi ifade etmeyen bir fonksiyon ismi. O yüzden `navigateToActivityWithExtras(...)` veya `startActivityWithExtras(...)` gibi daha ifade edici isimlendirme yapılabilir.

Ayrıca, tüm sayfalarda `command + option + L`(Mac) yaparak bazı basit formatlama ya da linting hatalarını giderebiliriz: 
Örnek: `destinationActivity: Activity,city : City ` burada virgülden sonra bir boşluk gelmeli. Aşağıda farklı bir yerde de böyle bir formatting hatası var.

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/util/Constant.kt:
```diff
@@ -0,0 +1,13 @@
+package com.example.secondtasksolution.util
+
+object Constant{
```
object Constant sonrasına bir boşluk bırakabilir misin

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode1/DetailActivity.kt:
```diff
@@ -0,0 +1,35 @@
+        binding = ActivityDetailBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        intent.apply {
```
intent içerisi okumayı farklı bir fonksiyona taşıyabiliriz

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode1/ListActivity.kt:
```diff
@@ -0,0 +1,41 @@
+        binding = ActivityListBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        val callback = object : OnBackPressedCallback(true) {
```
callback işlemini fonksiyona bölelim

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode1/ListActivity.kt:
```diff
@@ -0,0 +1,41 @@
+
+        onBackPressedDispatcher.addCallback(callback)
+
+        binding.listActivityRecyclerView.layoutManager = LinearLayoutManager(this)
```
liste ile alakalı işlemleri farklı bir fonksiyona taşıyalım

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode2/DetailFragment.kt:
```diff
@@ -0,0 +1,57 @@
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?){
+        super.onViewCreated(view, savedInstanceState)
+
+        val binding = FragmentDetailBinding.bind(view)
```
on create view içerisine taşıyalım

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode2/DetailFragment.kt:
```diff
@@ -0,0 +1,57 @@
+        val binding = FragmentDetailBinding.bind(view)
+        fragmentDetailBinding = binding
+
+        val callback = object : OnBackPressedCallback(true) {
```
farklı fonksiyona taşıyalım

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode2/DetailFragment.kt:
```diff
@@ -0,0 +1,57 @@
+
+        val bundle = arguments
+
+        bundle?.run {
```
farklı fonksiyona taşıyalım

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode2/FragmentToFragmentActivity.kt:
```diff
@@ -0,0 +1,29 @@
+        binding = ActivityFragmentToFragmentBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        val callback = object : OnBackPressedCallback(true) {
```
farklı fonksiyona taşıyalım

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode2/FragmentToFragmentActivity.kt:
```diff
@@ -0,0 +1,29 @@
+
+        onBackPressedDispatcher.addCallback(callback)
+
+        ListFragment().getFragment(this)
```
ListFragment.newInstance() olacak şekilde düzenleyelim

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/main/MainActivity.kt:
```diff
@@ -0,0 +1,35 @@
+        setContentView(binding.root)
+
+        with(binding){
+
```
boşlukları silelim

akndmr commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode2/ListFragment.kt:
```diff
@@ -0,0 +1,67 @@
+
+        requireActivity().onBackPressedDispatcher.addCallback(callback)
+
+        binding.listFragmentRecyclerView.layoutManager = LinearLayoutManager(context)
```
LayoutManager ile ilgili yapacağımız ekstra bir şey yok ise, direkt XML'den verebiliriz.
`layoutManager=""`

akndmr commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/model/City.kt:
```diff
@@ -0,0 +1,15 @@
+    val cityWeatherImage : Int,
+    var cityTemperature : Int,
+    val cityTemperatureMin : Int,
+    val cityTemperatureMax : Int
```
Kotlin trailing comma konusuna bakabilirsin. Şu an şart değil.

akndmr reviewed (commented):
Eline sağlık, Mustafa.
Sadece bir kaç dosyaya bakabildim. Kalanına müsait olunca bakarım.
Daha düzenli ve güzel görünüyor.
Bir kaç  ufak yorumum oldu.

scotoglu commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/view/episode3/ListNavFragment.kt:
```diff
@@ -0,0 +1,79 @@
+        val binding = FragmentListNavBinding.bind(view)
+        fragmentListNavBinding = binding
+
+        val callback = object : OnBackPressedCallback(true) {
```
Back aksiyonu için extension fonksiyonu yazabilir miyiz? `onBackPressed`, `handleBackPressed` olarak isimlendirilebilir. ActivityExt içerisine ekleyebilirsin. 

```
// Usage
onBackPressed {
  requireActivity().finish()
}
```

mustafakamber committed:
feat : ActivityController,CallbackHandler,FragmentController added
feat : Codes are divided into functions

mustafakamber committed:
feat : fragment navigated with instance

mustafakamber committed:
feat : spaces added

mustafakamber committed:
feat : space deleted

mustafakamber committed:
feat : add companion object in ListFragment & DetailFragment.
feat : scope functions updated
feat : binding operations updated
feat : intent and Bundle were used to transfer data, with the City class.

mustafakamber committed:
feat : popBackStack method added in DetailFragment

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/util/ActivityController.kt:
```diff
@@ -3,20 +3,15 @@ package com.example.secondtasksolution.util
-                    putExtra(CITY_WEATHER_NAME, city.weatherName)
-                    putExtra(CITY_WEATHER_IMAGE, city.weatherImage)
-                    putExtra(CITY_TEMPERATURE, city.temperature)
+                    putExtra(CITY_DATA,city)
```
boşluk bırakalım

iamkurtgoz commented on task-2-solution/SecondTaskSolution/app/src/main/java/com/example/secondtasksolution/adapter/CityAdapter.kt:
```diff
@@ -29,13 +29,13 @@ class CityAdapter(val cityList: MutableList<City>, val onItemSelected: (City) ->
             listRowTemperatureText.text = temperature.toString()
             listRowTemperatureRangeText.text =
-                temperatureMin.toString() + "° - " + temperatureMax.toString() + "°"
+                "$temperatureMin° - $temperatureMax°"
```
gereksiz boşluğu silelim

mustafakamber committed:
feat : spaces deleted in CityAdapter,
feat : spaces added before city

scotoglu reviewed (approved):
Eline sağlık
